FBSDKCoreKit/FBSDKCoreKit/FBSDKApplicationDelegate.m

```diff
@@ -420,6 +420,11 @@ - (void)_logIfAppLinkEvent:(NSURL *)url
     return;
   }
 
+  NSString * targetURLString = applinkData[@"target_url"];
+  if (targetURLString == nil || ![targetURLString isKindOfClass: [NSString class]]) {
+    return;
+  }
+
   NSURL *targetURL = [NSURL URLWithString:applinkData[@"target_url"]];
```

Comment:
@mangofever I have updated the code to include the change proposed above. Please review. Thanks.